pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -349,29 +404,31 @@ func getOptimalBlockIOForFile(path string) (*api.BlockIO, error) {
 	}, nil
 }
 
-func SetDriverCacheMode(disk *api.Disk) error {
+func SetDriverCacheMode(disk *api.Disk, checker DirectIOChecker) error {
 	var path string
 	supportDirectIO := true
 	mode := v1.DriverCache(disk.Driver.Cache)
+	isBlockDev := false
 
 	if disk.Source.File != "" {
 		path = disk.Source.File
 	} else if disk.Source.Dev != "" {
 		path = disk.Source.Dev
+		isBlockDev = true
 	} else {
 		return fmt.Errorf("Unable to set a driver cache mode, disk is neither a block device nor a file")
 	}
 
 	if mode == "" || mode == v1.CacheNone {
-		supportDirectIO = checkDirectIOFlag(path)
+		supportDirectIO = checkDirectIOFlag(path, isBlockDev, checker)
 		if !supportDirectIO {
 			log.Log.Infof("%s file system does not support direct I/O", path)
 		}
 		// when the disk is backed-up by another file, we need to also check if that
 		// file sits on a file system that supports direct I/O
 		if backingFile := disk.BackingStore; backingFile != nil {
 			backingFilePath := backingFile.Source.File
-			backFileDirectIOSupport := checkDirectIOFlag(backingFilePath)
+			backFileDirectIOSupport := checkDirectIOFlag(backingFilePath, false, checker)
```

Comment:
as a reader, this is still a confusion. I would be asking myself, what is so `false` here? I'd rather see here a simple call to `checker.CheckBlockDevice(path)`

Reply:
That is done like that mostly to avoid code duplication since `checkDirectIOFlag` already checks the error and prints a warning. I can do smth like that instead to make it clear:
```go
const isNotBlockDev = true
backFileDirectIOSupport := checkDirectIOFlag(backingFilePath, isNotBlockDev, checker)
```